src/librustc/query/mod.rs

```diff
@@ -505,6 +505,15 @@ rustc_queries! {
             desc { "extract field of const" }
         }
 
+        /// Destructure a constant ADT or array into its variant indent and its
+        /// field values.
+        query destructure_const(
+            key: ty::ParamEnvAnd<'tcx, &'tcx ty::Const<'tcx>>
+        ) -> mir::DestructuredConst<'tcx> {
```

Comment:
This can replace `const_field` (which IIRC is only used in a loop to get *all* the consts).
Also IMO it should have a `const_` prefix instead of suffix.

Reply:
I opened https://github.com/rust-lang/rust/issues/72264 for this